thorlcr/activities/lookupjoin/thlookupjoinslave.cpp

```diff
@@ -730,17 +740,13 @@ class CInMemJoinBase : public CSlaveActivity, public CThorDataLink, public CAllO
         ~CRowProcessor()
         {
             blockQueue.stop();
-            loop
-            {
-                Owned<CSendItem> sendItem = blockQueue.dequeueNow();
-                if (NULL == sendItem)
-                    break;
-            }
+            clearQueue();
             wait();
         }
         void start()
         {
             stopped = false;
+            clearQueue();
```

Comment:
should this be done in a RowProcessor::stop called from the activity::stop method?